### DISCUSSION TITLE
How should installing a library be handled?

### DISCUSSION THREAD

**Discussion by meator** [mesonbuild/meson#12838] [Q&A]
My program is using the `loguru` wrap, which provides a library with `library()`. It installs it and a header file. I'm not sure whether the header file should be installed but whatever.

The library must be installed alongside the program for the program to execute, because the library is dynamic.

Loguru is a small logging library, it doesn't make much sense for it to be a dynamic library, so I want a static library. I have passed `static: true` to `dependency()`. It is working well, but the library (with the header file) is still getting installed alongside my program (when invoking `meson install`). This is not tolerable behavior.

I see that the `loguru` wrap hasn't been built with static libraries in mind, but it should still be handled correctly, because Meson holds the responsibility for deciding which library is used. My question is, how can this be solved? Is the correct approach to check whether it's a static library and then make `install` depend on that? That's hacky. I can also patch out the installation using a diff file, but that is also not a great solution. It works around the problem.

**Comment by MartinSchwarzbauer** [comment]
Absolutly no idea, but try using 'static_library' instead of 'library'! Then only a static version is built!

**Reply by meator** [reply]
Getting a static library isn't really the problem. Using `static_library` is not possible in my situation, and it's also unnecessary, because that's what `static: true` does.

**Comment by bruchar1** [comment]
You can use `meson install --skip-subprojects` to only install files from your project (or you may skip specific subprojects).
You can also use `--tags` to only install `runtime` (shared library) files.
